src/js/components/FileInput/FileInput.js

```diff
@@ -239,16 +240,44 @@ const FileInput = forwardRef(
             onChange={event => {
               event.persist();
               const fileList = event.target.files;
+              let filesToAdd;
+              if (!files.length) {
+                if (
+                  multiple &&
+                  maximumCount &&
+                  fileList.length >= maximumCount
+                ) {
```

Comment:
Wouldn't this part be unreachable because if files is empty then fileList.length wouldn't be bigger than the maximumCount?

Reply:
Let me demostrate  using the video.

https://user-images.githubusercontent.com/61521805/114510404-a4959200-9c04-11eb-8fd3-2e3a286b5733.mp4